seed/lib/mappings/mapping_columns.py

```diff
@@ -163,35 +174,38 @@ def duplicates(self):
         duplicates = set()
         result = {}
 
-        for k, v in self.data.iteritems():
+        for raw_column, v in self.data.items():
             if v['initial_mapping_cmp'] in uniques:
                 duplicates.add(v['initial_mapping_cmp'])
             uniques.add(v['initial_mapping_cmp'])
 
         # now go through and populate the dict with the duplicate keys
         for item in duplicates:
-            for k, v in self.data.iteritems():
+            for raw_column, v in self.data.items():
                 if v['initial_mapping_cmp'] == item:
                     if item not in result.keys():
                         result[item] = []
                     result[item].append(
                         {
-                            'raw_column': k,
-                            'confidence': self.first_suggested_mapping(k)[2]
+                            'raw_column': raw_column,
+                            'confidence': self.first_suggested_mapping(raw_column)[2]
                         }
                     )
 
         return result
 
     def resolve_duplicate(self, dup_map_field, raw_columns):
         """
+        If there are duplicates, that is two raw_columns are trying to map to the same suggested
+        column, then select the next available one on the duplicate column. The one with the highest
+        confidence will 'win' the duplicate battle.
 
         :param dup_map_field: String, name of the field that is a duplicate
         :param columns: list, raw columns that mapped to the same result
         :return: None
 
         """
-        _log.debug("resolving duplicate field for {}".format(dup_map_field))
+        _log.debug("resolving duplicate field for %s" % dup_map_field)
```

Comment:
this is the actual fix to the unicode issue. Everything thing else were tests and code cleanup.